lib/Sema/ConstraintSystem.cpp

```diff
@@ -2839,14 +2834,61 @@ bool ConstraintSystem::diagnoseAmbiguityWithFixes(
                     diag::no_overloads_match_exactly_in_call_no_labels,
                     decl->getDescriptiveKind(), name.getBaseName());
       }
+    };
+
+    switch (ambiguityKind) {
+    case AmbiguityKind::CloseMatch:
+      // Handled below
+      break;
+    case AmbiguityKind::ParameterList: {
+      emitGeneralAmbiguityFailure();
+
+      for (const auto &viable: viableSolutions) {
+        auto overload = viable->getOverloadChoice(commonCalleeLocator);
+        auto *fn = overload.openedType->getAs<AnyFunctionType>();
+        assert(fn);
+        DE.diagnose(overload.choice.getDecl()->getLoc(),
```

Comment:
Just a question, I wonder if with this we might be able to get rid of the `wrong_argument_labels_overload` in `CSDiag` [here](https://github.com/apple/swift/blob/746b58e8e1c1ef1ac09d8031875fd2a08b65597c/lib/Sema/CSDiag.cpp#L1806)? I'm asking because I was looking into this just yesterday :))

Reply:
This change does cover ambiguity due to argument label mismatches so it's very possible that `wrong_argument_labels_overload` in CSDiag is obsolete. I'm testing it now!

Reply:
Seems to be obsolete - thanks for catching this!